arch/armv7a/jmp.S

```diff
@@ -6,90 +6,107 @@
  *
  * _setjmp, _longjmp, setjmp, sigsetjmp
  *
- * Copyright 2018 Phoenix Systems
- * Author: Kamil Amanowicz
+ * Copyright 2018, 2019 Phoenix Systems
+ * Author: Kamil Amanowicz, Aleksander Kaminski, Jakub Sarzynski
  *
  * This file is part of Phoenix-RTOS.
  *
  * %LICENSE%
  */
 
+ /* setjmp env buffer description
+  * Offset	Description
+  * 0		Signal mask flag - if != 0 then next value stores a signal mask
+  * 4		Signal mask
+  * 8		SP register
+  * 12-44	r4-r11 registers
+  * 44		LR register
+  * 48-80	d8-d15 registers
+  * 80		fpscr register
+  */
+
 #define __ASSEMBLY__
 
 .thumb
 .syntax unified
 
-.globl _setjmp
-.type _setjmp, %function
-_setjmp:
-	/* set jump address and stack pointer */
-	ldr r1, =(1f + 1)
-	mov r2, sp
-	/* store everythig in jmpbuf */
-	stm r0, {r1, r2, r4, r5, r6, r7, r8, r9, r10, r11, lr}
+
+.globl sigsetjmp
+.type sigsetjmp, %function
+sigsetjmp:
+	/* Store signal mask if savesigs != 0 */
+	mov r2, #0
```

Comment:
```suggestion
	movs r2, #0
```
2 bytes instead 4
same in line 44 
```
movs r0, #0
or 
mov r0, r2 /* r2 contains 0 */
```

Reply:
According to docs it shouldn't matter: https://developer.arm.com/documentation/ddi0308/d/Thumb-Instructions/Alphabetical-list-of-Thumb-instructions/MOV--immediate-
and I'd rather not use `s` here, as it makes the code somewhat confusing

Reply:
Such instructions are used by the compiler.
A simple example:
```c
typedef unsigned int jmp_buf[34];
typedef jmp_buf sigjmp_buf;

extern int sigsetjmp(sigjmp_buf env, int savesigs);

int _setjmp(jmp_buf var)
{
    return sigsetjmp(var, 0);
}
```
and disassembly
```
Disassembly of section .text._setjmp:

00000000 <_setjmp>:
   0:   2100            movs    r1, #0
   2:   b508            push    {r3, lr}
   4:   f7ff fffe       bl      0 <sigsetjmp>
   8:   bd08            pop     {r3, pc}
   a:   bf00            nop
```

Reply:
Ok, I see now. As it turns out there're only two 16-bit `mov` immediate instructions:
```
movs Rd, #imm8
mov<c> Rd, #imm8
```
There're no 16-bit `mov` instruction to be used outside IT block (https://developer.arm.com/documentation/dui0489/i/arm-and-thumb-instructions/mov). It all makes sense now to me, will change.